compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -863,13 +863,19 @@ OMR::ResolvedMethodSymbol::genAndAttachOSRCodeBlocks(int32_t currentInlinedSiteI
    // This exception edge should remain until OSR guards have been inserted, at which point it is no
    // longer needed
    //
-   bool genForOuterEntry = self()->comp()->isOutermostMethod() && self()->comp()->getHCRMode() == TR::osr;
-   if (genForOuterEntry)
+   if (self()->comp()->isOutermostMethod() && self()->comp()->getHCRMode() == TR::osr)
       {
+      // Add an exception edge to the OSR catch block from the first block
       TR::Block *OSRCatchBlock = osrMethodData->findOrCreateOSRCatchBlock(self()->getFirstTreeTop()->getNode());
       TR::Block *firstBlock = self()->getFirstTreeTop()->getEnclosingBlock();
       if (!firstBlock->hasExceptionSuccessor(OSRCatchBlock))
          self()->comp()->getFlowGraph()->addEdge(TR::CFGEdge::createExceptionEdge(firstBlock, OSRCatchBlock, self()->comp()->trMemory()));
+
+      // Create an OSR point with bytecode index 0
+      TR_ByteCodeInfo firstBCI = self()->getFirstTreeTop()->getNode()->getByteCodeInfo();
+      firstBCI.setByteCodeIndex(0);
```

Comment:
Okay so we have prepared for a transition at bytecode index 0 specifically in the IL gen which is what makes it the value you want to use here when creating an OSR point.